ui/app/pages/first-time-flow/create-password/import-with-seed-phrase/import-with-seed-phrase.component.js

```diff
@@ -234,12 +240,31 @@ export default class ImportWithSeedPhrase extends PureComponent {
         </div>
         <div className="first-time-flow__textarea-wrapper">
           <label>{ t('walletSeed') }</label>
-          <textarea
-            className="first-time-flow__textarea"
-            onChange={(e) => this.handleSeedPhraseChange(e.target.value)}
-            value={this.state.seedPhrase}
-            placeholder={t('seedPhrasePlaceholder')}
-          />
+          {hideSeedPhrase ? (
+            <TextField
+              className="first-time-flow__textarea"
+              type="password"
+              onChange={(e) => this.handleSeedPhraseChange(e.target.value)}
+              value={this.state.seedPhrase}
+              placeholder={t('seedPhrasePlaceholder')}
+            />
+          ) : (
+            <textarea
+              className="first-time-flow__textarea"
+              onChange={(e) => this.handleSeedPhraseChange(e.target.value)}
+              value={this.state.seedPhrase}
+              placeholder={t('seedPhrasePlaceholder')}
+            />
+          )}
```

Comment:
Nit: rather than repeat the `textarea` twice, you could conditionally set that single field instead:

```suggestion
          <TextField
            className="first-time-flow__textarea"
            type={ hideSeedPhrase ? 'password' : undefined }
            onChange={(e) => this.handleSeedPhraseChange(e.target.value)}
            value={this.state.seedPhrase}
            placeholder={t('seedPhrasePlaceholder')}
          />
```